sklearn/feature_selection/mutual_info.py

```diff
@@ -0,0 +1,293 @@
+        interpreted as the absolute number of features to select. The value
+        within (0.0, 1.0) is interpreted as the percentage from the initial
+        number of features (rounded down). Half of the features is selected by
+        default.
+
+    categorical_features : bool or array_like with shape (n_features),
+                           default False
+        If bool, then determines whether to consider all features categorical
+        or continuous. If array, then True elements mark categorical features.
+
+    categorical_target : bool, default False
+        Whether to consider `y` as a categorical variable.
+
+    n_neighbors : int, default 3
+        Number of neighbors to use for mutual information estimation for
+        continuous variables, see [2]_ and [3]_. The default value is usually
+        sufficient.
+
+    copy : bool, default True
+        Whether to make a copy of the given data. If set to False, the initial
+        data will be overwritten if it was in dense format.
+
+    Attributes
+    ----------
+    support_ : ndarray, shape (n_features,)
+        Mask of selected features.
+
+    relevance_ : ndarray, shape (n_features,)
+        Array containing mutual information values between each feature and
+        the target variable.
+
+    redundancy_ : ndarray, shape (n_features, n_features)
+        Array containing mutual information values between each pair of
+        features. Note, that diagonal elements are not computed and contain
+        some random numbers.
+
+    References
+    ----------
+    .. [1] H. Peng, F. Long, and C. Ding, "Feature selection based on mutual
+           information: criteria of max-dependency, max-relevance, and
+           min-redundancy", IEEE Transactions on Pattern Analysis and Machine
+           Intelligence, Vol. 27, No. 8, pp. 1226-1238, 2005.
+    .. [2] A. Kraskov, H. Stogbauer and P. Grassberger, "Estimating mutual
+           information". Phys. Rev. E 69, 2004.
+    .. [3] B. C. Ross "Mutual Information between Discrete and Continuous
+           Data Sets". PLoS ONE 9(2), 2014.
+    """
+    def __init__(self, n_features_to_select=0.5, categorical_features=False,
+                 categorical_target=False, n_neighbors=3, copy=True):
+        self.n_features_to_select = n_features_to_select
+        self.categorical_features = categorical_features
+        self.categorical_target = categorical_target
+        self.n_neighbors = n_neighbors
+        self.copy = copy
+
+    def _get_support_mask(self):
+        return self.support_
+
+    def fit(self, X, y):
+        """Find the best features according to mutual information criterion.
+
+        Parameters
+        ----------
+        X : array_like or sparse matrix, shape (n_samples, n_features)
+            Feature matrix.
+        y : array_like, shape (n_samples,)
+            Target vector.
+
+        Returns
+        -------
+        self
+        """
+        X, y = check_X_y(X, y, accept_sparse='csr',
```

Comment:
o.O This is odd indeed. I would expect it to be the other way around, no? since operations along the columns are easy in csc_matrices.
See: https://github.com/scikit-learn/scikit-learn/pull/5791